app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala

```diff
@@ -22,142 +22,163 @@
 import uk.gov.nationalarchives.omega.editorial.forms.{ EditSetRecordFormValues, EditSetRecordFormValuesFormProvider }
 import uk.gov.nationalarchives.omega.editorial.models._
 import uk.gov.nationalarchives.omega.editorial.services.CoveringDateCalculator.getStartAndEndDates
-import uk.gov.nationalarchives.omega.editorial.services.{ CoveringDateError, EditSetRecordService, ReferenceDataService }
+import uk.gov.nationalarchives.omega.editorial.services.{ CoveringDateError, EditSetRecordService, EditSetService, ReferenceDataService }
 import uk.gov.nationalarchives.omega.editorial.support.{ DateParser, FormSupport }
 import uk.gov.nationalarchives.omega.editorial.views.html.{ editSetRecordEdit, editSetRecordEditDiscard, editSetRecordEditSave }
-import uk.gov.nationalarchives.omega.editorial.{ controllers, editSetRecords, editSets }
 
-import scala.concurrent.{ ExecutionContext, Future }
 import java.time.LocalDate
 import java.time.temporal.ChronoField.{ DAY_OF_MONTH, MONTH_OF_YEAR, YEAR }
 import javax.inject.{ Inject, Singleton }
+import scala.concurrent.{ ExecutionContext, Future }
 
 @Singleton
 class EditSetRecordController @Inject() (
   messagesControllerComponents: MessagesControllerComponents,
   referenceDataService: ReferenceDataService,
+  editSetService: EditSetService,
   editRecordSetService: EditSetRecordService,
   editSetRecordEdit: editSetRecordEdit,
   editSetRecordEditDiscard: editSetRecordEditDiscard,
   editSetRecordEditSave: editSetRecordEditSave
-)(implicit val ec: ExecutionContext)
-    extends MessagesAbstractController(messagesControllerComponents) with I18nSupport with Secured with FormSupport {
+)(implicit ec: ExecutionContext)
+    extends BaseAppController(messagesControllerComponents, editSetService, editRecordSetService) {
 
   private val logger: Logger = Logger(this.getClass)
   private lazy val legalStatuses: Seq[LegalStatus] = referenceDataService.getLegalStatuses
   private lazy val creators: Seq[Creator] = referenceDataService.getCreators
   private lazy val placesOfDeposit: Seq[PlaceOfDeposit] = referenceDataService.getPlacesOfDeposit
 
-  def viewEditRecordForm(id: String, recordId: String): Action[AnyContent] = Action.async {
+  def viewEditRecordForm(editSetId: String, recordId: String): Action[AnyContent] = Action.async {
     implicit request: Request[AnyContent] =>
       withUserAsync { user =>
-        logger.info(s"The edit set id is $id for record id $recordId")
-        val editSetName = editSets.getEditSet().name
-        findRecord(id, recordId).map {
-          case Right(record) =>
-            val recordPreparedForDisplay = prepareForDisplay(record)
-            val recordForm = bindFormFromRecordForDisplay(recordPreparedForDisplay)
-            Ok(generateEditSetRecordEditView(user, editSetName, recordPreparedForDisplay, recordForm))
-          case Left(_) => NotFound
+        logger.info(s"The edit set id is $editSetId for record id $recordId")
+        findEditSet(editSetId).flatMap {
```

Comment:
I think these nested futures might be better expressed as a for comprehension, since the error type extends from the same superclass.

Reply:
I'll try that.

Reply:
Changed.

Reply:
Resolving as @felixgb approved the PR.